blocks/rich-text/index.js

```diff
@@ -558,6 +559,37 @@ export class RichText extends Component {
 		if ( keyCode === BACKSPACE ) {
 			this.onChange();
 		}
+
+		// `scrollToCaret` is called on `nodechange`, whereas calling it on
+		// `keyup` *when* moving to a new RichText element results in incorrect
+		// scrolling. Though the following allows false positives, it results
+		// in much smoother scrolling.
+		if ( keyCode !== BACKSPACE && keyCode !== ENTER ) {
+			this.scrollToCaret();
+		}
+	}
+
+	scrollToCaret() {
+		const { top: caretTop } = this.getEditorSelectionRect();
+
+		const container = getScrollContainer( this.editor.getBody() );
+
+		// When scrolling, avoid positioning the caret at the very top of
+		// the viewport, providing some "air" and some textual context for
+		// the user, and avoiding toolbars.
+		const graceOffset = this.props.isViewportSmall ? 100 : 300;
```

Comment:
I wasn't very with reading `clientHeight` on every key release, and also with the fact that it was blindly grabbing a third of the height when we don't know what the device actually looks like, so I went with this.

Reply:
I was trying to find a reference, but maybe I dreamt the whole thing. I thought I recalled Paul Lewis from Google, who often talks about performance, bringing up certain read interfaces of the DOM that were more expensive than one would suspect. I was looking [here](https://developers.google.com/web/fundamentals/performance/rendering/) but didn't find anything. Maybe it's totally fine.

Anyway, per my other comment about rolling this out for mobile only, do you think it's still worth having `clientHeight / 3.0`, or would the magic `100` be enough?